drivers/sensor/bmi160/bmx160_i2c.c

```diff
@@ -0,0 +1,67 @@
+/* Bosch BMX160 inertial measurement unit driver
+ *
+ * Copyright (c) 2020 Linumiz
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Datasheet:
+ * https://ae-bst.resource.bosch.com/media/_tech/media/datasheets/BST-BMX160-DS000.pdf
+ */
+
+#include <init.h>
+#include <drivers/sensor.h>
+#include <sys/byteorder.h>
+#include <kernel.h>
+#include <sys/__assert.h>
+#include <logging/log.h>
+#include "bmi160.h"
+
+#define BMX160_CHIP_ID		0xD8
+#define DT_DRV_COMPAT		bosch_bmx160
+
+LOG_MODULE_REGISTER(BMX160, CONFIG_SENSOR_LOG_LEVEL);
+
+struct bmi160_device_data bmx160_data = {
+	.chipid = BMX160_CHIP_ID,
+};
+
+static const struct sensor_driver_api bmx160_api = {
+	.attr_set = bmi160_attr_set,
+#ifdef CONFIG_BMI160_TRIGGER
+	.trigger_set = bmi160_trigger_set,
+#endif
+	.sample_fetch = bmi160_sample_fetch,
+	.channel_get = bmi160_channel_get,
+};
+
+int bmx160_init(struct device *dev)
+{
+	struct bmi160_device_data *bmx160 = dev->driver_data;
+	int ret;
+
+	bmx160->i2c = device_get_binding(DT_INST_BUS_LABEL(0));
+	if (!bmx160->i2c) {
+		LOG_ERR("I2C master controller not found: %s.",
+			    DT_INST_BUS_LABEL(0));
+		return -EINVAL;
+	}
+
+	bmx160->i2c_addr = DT_INST_REG_ADDR(0);
+	ret =  bm160_device_init(dev);
+	if (ret)
+		LOG_ERR("device init fail: %d", ret);
+
+	return ret;
+}
+
+const struct bmi160_device_config bmx160_config = {
+#if defined(CONFIG_BMI160_TRIGGER)
+	.gpio_port = DT_INST_GPIO_LABEL(0, int_gpios),
+	.int_pin = DT_INST_GPIO_PIN(0, int_gpios),
+	.int_flags = DT_INST_GPIO_FLAGS(0, int_gpios),
+#endif
+};
+
+DEVICE_AND_API_INIT(bmx160, DT_INST_LABEL(0), bmx160_init, &bmx160_data,
+		&bmx160_config, POST_KERNEL, CONFIG_SENSOR_INIT_PRIORITY,
+		&bmx160_api);
```

Comment:
Driver should support an arbitrary number of instances, see `DT_INST_FOREACH` examples.